examples/step-62/step-62.cc

```diff
@@ -0,0 +1,1467 @@
+  // mirrors and a $\lambda/2$ cavity where $\lambda$ is the acoustic
+  // wavelength. Acoustic DBRs are  periodic structures where a set of bilayer
+  // stacks with contrasting physical properties (sound velocity index) is
+  // repeated $N$ times. The change of in the velocity will be obtained by
+  // alternating layers with different density.
+  template <int dim>
+  Rho<dim>::Rho(HDF5::Group &data)
+    : Function<dim>(1)
+    , data(data)
+    , lambda(data.get_attribute<double>("lambda"))
+    , mu(data.get_attribute<double>("mu"))
+    , material_a_rho(data.get_attribute<double>("material_a_rho"))
+    , material_b_rho(data.get_attribute<double>("material_b_rho"))
+    , cavity_resonance_frequency(
+        data.get_attribute<double>("cavity_resonance_frequency"))
+    , nb_mirror_pairs(data.get_attribute<int>("nb_mirror_pairs"))
+    , dimension_y(data.get_attribute<double>("dimension_y"))
+    , grid_level(data.get_attribute<int>("grid_level"))
+  {
+    // In order to increase the precision we use
+    // [subpixel
+    // smoothing](https://meep.readthedocs.io/en/latest/Subpixel_Smoothing/).
+    average_rho_width = dimension_y / (std::pow(2.0, grid_level));
+    data.set_attribute("average_rho_width", average_rho_width);
+  }
+
+
+
+  template <int dim>
+  double Rho<dim>::value(const Point<dim> &p,
+                         const unsigned int /*component*/) const
+  {
+    // The speed of sound is defined by
+    // @f[
+    //  c = \frac{K_e}{\rho}
+    // @f]
+    // where $K_e$ is the effective elastic constant and $\rho$ the density.
+    // Here we consider the case in which the waveguide width is much smaller
+    // than the wavelength. In this case it can be shown that for a two
+    // dimensional case
+    // @f[
+    //  K_e = 4\mu\frac{\lambda +\mu}{\lamda+2\mu}
+    // @f]
+    // and for a three dimensional case $K_e$ is equal to the Young's modulus.
+    // @f[
+    //  K_e = 4\mu\frac{\lambda +\mu}{\lamda+2\mu}
+    // @f]
+    double elastic_constant;
+    if (dim == 2)
+      {
+        elastic_constant = 4 * mu * (lambda + mu) / (lambda + 2 * mu);
+      }
+    else if (dim == 3)
+      {
+        elastic_constant = mu * (3 * lambda + 2 * mu) / (lambda + mu);
+      }
+    else
+      {
+        Assert(false, ExcInternalError());
+      }
+    const double material_a_speed_of_sound =
+      std::sqrt(elastic_constant / material_a_rho);
+    const double material_a_wavelength =
+      material_a_speed_of_sound / cavity_resonance_frequency;
+    const double material_b_speed_of_sound =
+      std::sqrt(elastic_constant / material_b_rho);
+    const double material_b_wavelength =
+      material_b_speed_of_sound / cavity_resonance_frequency;
+
+    // The density $\rho$ takes the following form
+    //<img alt="Phononic superlattice cavity"
+    // src="https://raw.githubusercontent.com/dangars/dealii/phononic-cavity/examples/step-62/doc/step-62.04.svg?sanitize=true"
+    // height="200" />
```

Comment:
Before this program is committed, we should change these links to point to the deal.II image repository. If you send me all of the pictures you'd like to include in this program, then I can add them there and give you links to reference them!

Reply:
I just sent you an email :)
I'll update the links and remove the commit with the images, when they'll be in the deal.II image repository.